src/win32_window.c

```diff
@@ -2375,13 +2401,32 @@ const char* _glfwGetClipboardStringWin32(void)
 
 void _glfwSetThemeWin32(_GLFWwindow* window, _GLFWtheme* theme)
 {
-    _glfwInputError(GLFW_FEATURE_UNIMPLEMENTED, NULL);
+    if (!theme || theme->variation == GLFW_THEME_DEFAULT)
+    {
+        applySystemTheme(window->win32.handle);
+        return;
+    }
+
+    GLFWbool darkMode = theme->variation == GLFW_THEME_DARK;
+
+    DwmSetWindowAttribute(window->win32.handle,
+                          DWMWA_USE_IMMERSIVE_DARK_MODE,
+                          &darkMode,
+                          sizeof(darkMode));
```

Comment:
In the same way that it might be possible to get the user-specified theme colour, is it also possible to set it, for a window? For instance, the title bar colours, the border colour, etc.

Reply:
Those colors are set by the system automatically depending on the system settings.
<img width="484" alt="image" src="https://user-images.githubusercontent.com/28705694/220502168-85f2568d-ccf9-4b22-886f-a3a3f988c4f0.png">
So, I suppose it might not be necessary to manually set them. Unless there're cases you want a different color from the system settings.

Reply:
Yes, that's exactly what I planned to let the GLFW users do. The default behaviour of `GLFWtheme` in its current design, is for its `GLFW_THEME_COLOR_MAIN` attribute (flag) to be `GLFW_FALSE` upon creation with `glfwCreateTheme`. If the user calls `glfwThemeSetAttribute(theme, GLFW_THEME_COLOR_MAIN , GLFW_TRUE)` (must be followed with `glfwThemeSetColor(theme, ?, ?, ?, ?)`, GLFW is supposed to apply that color to the window when the user calls `glfwSetTheme(window, theme)`.

Reply:
I planned for this to be supposed usage of the API, but in hindsight, it leads to problems with default vs user-specified attributes.

```
GLFWwindow* window = ...;
GLFWtheme* theme = glfwGetTheme(window);

glfwThemeSetAttribute(theme, GLFW_THEME_COLOR_MAIN , GLFW_TRUE);
glfwThemeSetColor(theme, GLFW_THEME_COLOR_MAIN, ..., ..., ..., ...);

glfwSetTheme(window, theme);
```

This does not work as intended with the current implementation. The reason is that `glfwGetTheme` defaults to return the actual theme attributes of the window.

Consider this:

```
GLFWwindow* window = ...;
glfwSetTheme(window, glfwGetTheme(window));
```

If the window is set to use the default system theme and color, `glfwGetTheme` does not return a theme object with these properties set to default (`GLFW_THEME_DEFAULT` for the variation, `GLFW_FALSE` for the colour). Instead, it polls the default from the system, and sets the theme's variation to the current variation of the window (as specified by the system), and if the system provides access to a colour, sets that on the theme too.
So the code snippet above, will change a window's theme from the system defaults (and automatically responding to system changes), to the currently set system theme, and no longer automatically update.

Reply:
So, if only considering the light/dark now:

When a window is created, it'll use the system default brightness. When the system changes its mode, the window's mode will change to.

However, if `glfwSetTheme(window, glfwGetTheme(window))`, the window will no longer change its mode when the system does. This is not intuitive, and is a big fault in my design. The reason I did it, however, was to let users query the current actual values of the window with a call to `glfwGetTheme`.

Reply:
Any chance you could also add a `setAccentColor`? Is that feasible?

Reply:
@ws909  I'm not sure if that's possible, but I'll have a try.